src/mono/mono/mini/mini-llvm.c

```diff
@@ -5218,7 +5218,32 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
 			break;
 		}
 
-		case OP_SETRET:
+		case OP_SETRET: {
+			gboolean skip = FALSE;
+			switch (linfo->ret.storage) {
+			case LLVMArgNone:
+			case LLVMArgNormal:
+			case LLVMArgVtypeInReg: {
+				if (!lhs || ctx->is_dead [ins->sreg1]) {
+					/*
+					 * The method did not set its return value, probably because it
+					 * ends with a throw.
+					 */
```

Comment:
Can it then emit an `unreachable` instruction here?

Reply:
We already do that i think when emitting the throw.

Reply:
This is about values [ins->sreg1] not being set since the instruction which is supposed to set it was not emitted.